src/table.ts

```diff
@@ -23,44 +23,35 @@ import * as through from 'through2';
 import {Operation as GaxOperation} from 'google-gax';
 import {Json} from './codec';
 import {Database} from './database';
-import {SpannerClient as s} from './v1';
 import {PartialResultStream, Row} from './partial-result-stream';
-import {ReadRequest, TimestampBounds} from './transaction';
+import {
+  ReadRequest,
+  TimestampBounds,
+  CommitResponse,
+  ReadResponse,
+  ReadCallback,
+  Transaction,
+  CommitCallback,
+} from './transaction';
 import {google as databaseAdmin} from '../proto/spanner_database_admin';
-import {Schema, ResourceCallback} from './common';
+import {Schema, ResourceCallback, LongRunningCallback} from './common';
 
 export type Key = string | string[];
 
-type CommitPromise = Promise<[s.CommitResponse]>;
 export type CreateTableResponse = [
   Table,
   GaxOperation,
   databaseAdmin.longrunning.IOperation
 ];
-type DropTablePromise = Promise<
-  [GaxOperation, databaseAdmin.longrunning.IOperation]
->;
-type ReadPromise = Promise<[Array<Row | Json>]>;
+type DropTableResponse = [GaxOperation, databaseAdmin.longrunning.IOperation];
```

Comment:
Suggesting to utilize `UpdateSchemaCallback/Response` as this method is a wrapper around `Database.updateSchema`
```suggestion
export type DropTableResponse = UpdateSchemaResponse;
```